zon.stringify: tagged unions with void fields do not serialize

### Zig Version

0.14.0-dev.3241+55c46870b

### Steps to Reproduce and Observed Behavior

`mwe.zig`:
```zig
 pub const foo = union(enum) {
    bar,
};

pub fn main() !void {
    const t: foo = .bar;

    std.zon.stringify.serialize(t, .{}, std.io.null_writer) catch unreachable;
}
const std = @import("std");
```

run: `zig run mwe.zig`

Observe:
```
zig run mwe.zig
/nix/store/4h7cpfzzsczh6g8ivdvaz5k9dygy5z6p-zig-0.14.0-dev.3241+55c46870b/lib/std/debug.zig:518:14: error: reached unreachable code
    if (!ok) unreachable; // assertion failure
             ^~~~~~~~~~~
/nix/store/4h7cpfzzsczh6g8ivdvaz5k9dygy5z6p-zig-0.14.0-dev.3241+55c46870b/lib/std/zon/stringify.zig:493:28: note: called from here
            comptime assert(canSerializeType(@TypeOf(val)));
                     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    fieldArbitraryDepth__anon_6382: /nix/store/4h7cpfzzsczh6g8ivdvaz5k9dygy5z6p-zig-0.14.0-dev.3241+55c46870b/lib/std/zon/stringify.zig:1005:56
    fieldArbitraryDepth__anon_4262: /nix/store/4h7cpfzzsczh6g8ivdvaz5k9dygy5z6p-zig-0.14.0-dev.3241+55c46870b/lib/std/zon/stringify.zig:907:55
    9 reference(s) hidden; use '-freference-trace=11' to see all references
```

### Expected Behavior

A union with a void field should serialize

--- a/lib/std/zon/stringify.zig
+++ b/lib/std/zon/stringify.zig
@@ -504,7 +504,6 @@ pub fn Serializer(Writer: type) type {
                 .bool, .null => try std.fmt.format(self.writer, "{}", .{val}),
                 .enum_literal => try self.ident(@tagName(val)),
                 .@"enum" => try self.ident(@tagName(val)),
-                .void => try self.writer.writeAll("{}"),
                 .pointer => |pointer| {
                     // Try to serialize as a string
                     const item: ?type = switch (@typeInfo(pointer.child)) {
@@ -579,15 +578,21 @@ pub fn Serializer(Writer: type) type {
                 },
                 .@"union" => |@"union"| {
                     comptime assert(@"union".tag_type != null);
-                    var container = try self.startStruct(.{ .whitespace_style = .{ .fields = 1 } });
                     switch (val) {
-                        inline else => |pl, tag| try container.fieldArbitraryDepth(
-                            @tagName(tag),
-                            pl,
-                            options,
-                        ),
+                        inline else => |pl, tag| if (@TypeOf(pl) == void)
+                            try self.writer.print(".{s}", .{@tagName(tag)})
+                        else {
+                            var container = try self.startStruct(.{ .whitespace_style = .{ .fields = 1 } });
+
+                            try container.fieldArbitraryDepth(
+                                @tagName(tag),
+                                pl,
+                                options,
+                            );
+
+                            try container.finish();
+                        },
                     }
-                    try container.finish();
                 },
                 .optional => if (val) |inner| {
                     try self.valueArbitraryDepth(inner, options);
@@ -1116,6 +1121,12 @@ test "std.zon stringify whitespace, high level API" {
         \\    3,
         \\} }
     , .{ .inner = .{ 1, 2, 3 } }, .{});
+
+    const UnionWithVoid = union(enum) { a, b: void, c: u8 };
+
+    try expectSerializeEqual(
+        \\.a
+    , UnionWithVoid.a, .{});
 }
 
 test "std.zon stringify whitespace, low level API" {

